app/javascript/components/subject/ExtendIsolation.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Form, Button, Modal, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import _ from 'lodash';
+import axios from 'axios';
+import moment from 'moment';
+
+import DateInput from '../util/DateInput';
+import reportError from '../util/ReportError';
+
+class ExtendIsolation extends React.Component {
```

Comment:
I'm having trouble differentiating the need for two components. Why not just handle everything in one place? Seems like the better option would be to do it in the datepicker area you added, and drop the button.

Reply:
One of the components is for the "Extend Isolation" button and the other one is for just the date picker on the bottom. I agree that if we were to only choose one of them, the datepicker would be the one to keep. In the group chat with Kelly though, it seemed like she wanted both, so we would have to double check with her?

Reply:
Yes please (via Slack)